@navikt/core/react/src/form/combobox/Combobox.tsx

```diff
@@ -0,0 +1,247 @@
+      if (selectedOptions.includes(value.trim())) return;
+      addCustomOption({ value });
+      handleClear(event);
+      focusInput();
+    },
+    [selectedOptions, value, addCustomOption, handleClear, focusInput]
+  );
+
+  const toggleOption = useCallback(
+    (event) => {
+      const clickedOption = event?.target?.textContent;
+      const focusedOption = currentOption;
+      // onClick: toggle selected option
+      if (clickedOption) {
+        if (selectedOptions.includes(clickedOption)) {
+          handleDeleteSelectedOption(clickedOption);
+        } else if (filteredOptions.includes(clickedOption))
+          addSelectedOption(clickedOption);
+      }
+      // onEnter: remove selected filteredOption
+      else if (focusedOption && selectedOptions.includes(focusedOption)) {
+        removeSelectedOption(focusedOption);
+        if (customOptions.includes(focusedOption))
+          removeCustomOption({ value: focusedOption });
+      }
+      // onEnter: add focused option
+      else if (focusedOption && filteredOptions?.includes?.(focusedOption)) {
+        addSelectedOption(focusedOption);
+      }
+      //onEnter: add custom option
+      else if (focusedOption && !filteredOptions.includes(focusedOption)) {
+        handleAddCustomOption(event);
+      }
+    },
+    [
+      currentOption,
+      selectedOptions,
+      filteredOptions,
+      addSelectedOption,
+      handleDeleteSelectedOption,
+      removeSelectedOption,
+      customOptions,
+      removeCustomOption,
+      handleAddCustomOption,
+    ]
+  );
+
+  //focus on input whenever selectedOptions changes
+  useEffect(() => {
+    if (prevSelectedOptions !== selectedOptions) focusInput();
+  }, [focusInput, selectedOptions, prevSelectedOptions]);
+
+  useEffect(() => {
+    if (shouldAutocomplete && inputRef && value !== searchTerm) {
+      inputRef.current?.setSelectionRange?.(searchTerm.length, value.length);
+    }
+  }, [value, inputRef, searchTerm, shouldAutocomplete]);
+
+  return (
+    <ComboboxWrapper
+      className={className}
+      hasError={hasError}
+      inputProps={inputProps}
+      inputSize={size}
+      toggleIsListOpen={toggleIsListOpen}
+    >
+      <Label
+        htmlFor={inputProps.id}
+        size={size}
+        className={cl("navds-form-field__label", {
+          "navds-sr-only": hideLabel,
+        })}
+      >
+        {label}
+      </Label>
+      {!!description && (
+        <BodyShort
+          as="div"
+          className={cl("navds-form-field__description", {
+            "navds-sr-only": hideLabel,
+          })}
+          id={inputDescriptionId}
+          size={size}
+        >
+          {description}
+        </BodyShort>
+      )}
+      <div className="navds-combobox__wrapper">
+        <div
+          className={cl(
+            "navds-combobox__wrapper-inner navds-text-field__input",
+            {
+              "navds-combobox__wrapper-inner--virtually-unfocused":
+                currentOption,
```

Comment:
Som edgecase kan `currentOption` være `""` -> `Boolean("") === false` 
```suggestion
                !!currentOption,
```